core/testcontainers/core/image.py

```diff
@@ -0,0 +1,115 @@
+import functools as ft
+from typing import Optional
+
+import docker
+from docker.client import DockerClient
+from docker.models.images import Image, ImageCollection
+
+from .utils import setup_logger
+
+LOGGER = setup_logger(__name__)
+
+
+class DockerImage:
+    """
+    Basic class to manage docker images.
+
+    .. doctest::
+            >>> from testcontainers.core.image import DockerImage
+
+            >>> image = DockerImage().from_dockerfile(path="core/tests/", tag="testcontainers/test-image")
+            >>> image.exists("testcontainers/test-image")
+            True
+            >>> image.get("testcontainers/test-image").id
+            'sha256:...'
+            >>> image.remove(force=True)
+            >>> image.exists("testcontainers/test-image")
+            False
+    """
+
+    def __init__(self, docker_client_kw: Optional[dict] = None, **kwargs) -> None:
+        self._docker = DockerClient().from_env(**(docker_client_kw or {}))
+
+    def from_dockerfile(self, path: str, tag: str = "local/image") -> "DockerImage":
```

Comment:
lets not make it dependent on an image, it should be possible to just use the image id, i think, both based on the java implementation [link](https://github.com/testcontainers/testcontainers-java/blob/d3b0df2d89ee349890c94b0ce220a167c82ddf51/core/src/main/java/org/testcontainers/images/builder/ImageFromDockerfile.java#L163) and also based on experience running `docker build` and forgetting to specify `-t myimage` - you can grab the id and tag with a separate command. not sure how feasible this is but without this, seems like it would be quite a limitation on concurrency...